From 6eea251ade8c142a80eec9c00a11db1eb0753b58 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Tue, 30 Apr 2013 11:48:11 +0100
Subject: [PATCH] journal: Set the default keep free value to 15% (up from 5%)

As some SSDs are still seeing performance degredation when
reaching 85% usage the default value of 5% seems a little low.

Set this to 15% by default.
(cherry picked from commit 8621b1109b0bbe097f2b3bfce2211fe8474c567b)
---
 man/journald.conf.xml      | 2 +-
 src/journal/journal-file.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/man/journald.conf.xml b/man/journald.conf.xml
index 70b05f6..0fb5d79 100644
--- a/man/journald.conf.xml
+++ b/man/journald.conf.xml
@@ -246,7 +246,7 @@
                                 configured in
                                 <varname>SystemMaxUse=</varname> and
                                 <varname>RuntimeMaxUse=</varname> is
-                                available. Defaults to 5% of the size
+                                available. Defaults to 15% of the size
                                 of the respective file
                                 system. <varname>SystemMaxFileSize=</varname>
                                 and
diff --git a/src/journal/journal-file.c b/src/journal/journal-file.c
index 3df099d..6eb3098 100644
--- a/src/journal/journal-file.c
+++ b/src/journal/journal-file.c
@@ -2731,7 +2731,7 @@ void journal_default_metrics(JournalMetrics *m, int fd) {
         if (m->keep_free == (uint64_t) -1) {
 
                 if (fs_size > 0) {
-                        m->keep_free = PAGE_ALIGN(fs_size / 20); /* 5% of file system size */
+                        m->keep_free = PAGE_ALIGN(fs_size * 3 / 20); /* 15% of file system size */
 
                         if (m->keep_free > DEFAULT_KEEP_FREE_UPPER)
                                 m->keep_free = DEFAULT_KEEP_FREE_UPPER;
-- 
1.8.1.5

